### PR TITLE
Add a minimal dockerfile

### PR DESCRIPTION
**Pull request by quentinlesceller**
Fix https://github.com/mimblewimble/grin/issues/919 by adding a minimal dockerfile.

Build instructions:
```
$ docker build -t grin .
$ docker run -it --rm --name grin_node grin
```

Still have weird character with the TUI despite having UTF-8 locales.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+RUN cargo build --release
+
+# runtime stage
+FROM debian:9.4
+
+RUN set -ex && \
+    apt-get update && \
+    apt-get --no-install-recommends --yes install locales && \
+    apt-get clean && \
+    rm -rf /var/lib/apt && \
```
Comment (ignopeverell) [inline Dockerfile]: Just curious, why remove all apt data here? Smaller image file?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+# Building Grin
+RUN cargo build --release
+
+# runtime stage
+FROM debian:9.4
+
+RUN set -ex && \
+    apt-get update && \
+    apt-get --no-install-recommends --yes install locales && \
```
Comment (bladedoyle) [inline Dockerfile]: Dont we still need ncurses library in the runtime image?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    apt-get clean && \
+    rm -rf /var/lib/apt && \
+    sed -i '157 s/^##*//' /etc/locale.gen && \
+    locale-gen
+
+COPY --from=builder /usr/src/grin/target/release/grin /usr/local/bin/grin
+COPY --from=builder /usr/src/grin/grin.toml /usr/src/grin/grin.toml
```
Comment (bladedoyle) [inline Dockerfile]: Do we want to disable the TUI for running in a container?

File: Dockerfile (hunk 1 above)
Comment (quentinlesceller) [reply Dockerfile]: Yes, however I'm updating this part and we will not need to remove the apt files.

File: Dockerfile (hunk 2 above)
Comment (quentinlesceller) [reply Dockerfile]: It looks like we don't need ncurses in the runtime image (tested locally on macOS 10.13.5).

File: Dockerfile (hunk 3 above)
Comment (quentinlesceller) [reply Dockerfile]: Currently when you do ```docker run -it grin_app```it launch just like ```grin server run```.  This is just a really minimal docker image.

**Comment by quentinlesceller** [comment]
Okay merging that now.

File: Dockerfile (hunk 3 above)
Comment (quentinlesceller) [reply Dockerfile]: Well actually, I'm leaving the TUI as it's really nice to be able to run in foreground with TUI.

**Pull request merged**